packages/records/lib/models/records.ts

```diff
@@ -165,12 +165,8 @@ export async function getRecords({
                 return Err(error);
             }
 
-            query = query.where(
-                (builder) =>
-                    void builder
-                        .where('updated_at', '>', decodedCursor.sort)
-                        .orWhere((builder) => void builder.where('updated_at', '=', decodedCursor.sort).andWhere('id', '>', decodedCursor.id))
-            );
+            // Tuple comparison for efficient index usage
+            query = query.whereRaw('(updated_at, id) > (?, ?)', [decodedCursor.sort, decodedCursor.id]);
```

Comment:
Does this not lose the second part of the condition by only checking > (or I'm misunderstanding something)?
before: `updated_at > decodedCursor.sort OR (updated_at = decodedCursor.sort AND id > decodedCursor.id)`
after: `updatedAt > decodedCursor.sort AND id > decodedCursor.id`

Reply:
from pg docs: https://www.postgresql.org/docs/current/functions-comparisons.html

> For the <, <=, > and >= cases, the row elements are compared left-to-right, stopping as soon as an unequal or null pair of elements is found... For example, ROW(1,2,NULL) < ROW(1,3,0) yields true...

in our case pg compare first the `updatedAt < decodedCursor.sort`, if false return, if true move to the second comparison (the `id`)